### PR TITLE
repair: coroutinize repair_range

### PR DESCRIPTION
**Pull request by avikivity**
Nicer and simpler, but essentially cosmetic.

### CODE REVIEW

File: repair/repair.cc [hunk 1]
```diff
@@ -647,7 +649,7 @@ future<> shard_repair_task_impl::repair_range(const dht::token_range& range, ::t
                 rlogger.error("repair[{}]: Repair {} out of {} ranges, shard={}, keyspace={}, table={}, range={}, peers={}, live_peers={}, status={}",
                         id.uuid(), ranges_index, ranges_size(), id.shard(), _status.keyspace, table_names(), range, neighbors, live_neighbors, status);
                 abort_repair_info();
-                return make_exception_future<>(std::runtime_error(format("Repair mandatory neighbor={} is not alive, keyspace={}, mandatory_neighbors={}",
+                co_await coroutine::return_exception(std::runtime_error(format("Repair mandatory neighbor={} is not alive, keyspace={}, mandatory_neighbors={}",
```
Comment (denesb) [inline repair/repair.cc]: Didn't this convert an earlier return to a simple `co_await`?

Comment (avikivity) [reply repair/repair.cc]: `co_await return_exception` actually returns. It's very tricky but we couldn't find better syntax.

Comment (avikivity) [reply repair/repair.cc]: /cc @piodul to confirm.

Comment (piodul) [reply repair/repair.cc]: I confirm. We previously had `co_return make_exception` (now deprecated) but it didn't work in coroutines that return void futures due to language limitations. `co_await return_exception` looks hacky but works for all return types.

**Review by denesb — APPROVED** [review]

**Pull request closed without merging**